pkg/operator/ceph/cluster/nodedaemon/exporter.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodedaemon
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strconv"
+
+	"github.com/pkg/errors"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/config"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+
+	cephver "github.com/rook/rook/pkg/operator/ceph/version"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	monitoringPath                   = "/etc/ceph-monitoring/"
+	serviceMonitorFile               = "ceph-exporter-service-monitor.yaml"
```

Comment:
Looks like the exporter daemon is failing to start because of this
```suggestion
	serviceMonitorFile               = "exporter-service-monitor.yaml"
```

Reply:
After I fixed this issue and tested again locally, I see this error in the exporter pod log:
```
2023-01-24T00:23:09.914+0000 7fa024b26f00 -1 asok(0x558e6c6d6560) AdminSocketConfigObs::init: failed: AdminSocket::bind_and_listen: failed to bind the UNIX domain socket to '/var/run/ceph/ceph-client.admin.asok': (17) File existssystem:0
```
Are you able to test it locally (e.g. in minikube)? CI needs to be green as well as confirm with manual testing.

Reply:
> After I fixed this issue and tested again locally, I see this error in the exporter pod log:
> 
> ```
> 2023-01-24T00:23:09.914+0000 7fa024b26f00 -1 asok(0x558e6c6d6560) AdminSocketConfigObs::init: failed: AdminSocket::bind_and_listen: failed to bind the UNIX domain socket to '/var/run/ceph/ceph-client.admin.asok': (17) File existssystem:0
> ```
> 
> Are you able to test it locally (e.g. in minikube)? CI needs to be green as well as confirm with manual testing.

Yes, it works fine for me locally with minikube setup
```csi-cephfsplugin-khxvr                              2/2     Running     0             119s
csi-cephfsplugin-provisioner-5b6959d889-tbk2m       5/5     Running     0             119s
csi-rbdplugin-8lfkt                                 2/2     Running     0             119s
csi-rbdplugin-provisioner-d6df8c994-6pf2l           5/5     Running     0             119s
prometheus-rook-prometheus-0                        3/3     Running     1 (98s ago)   2m10s
rook-ceph-crashcollector-minikube-59ff66b49-hr7j4   1/1     Running     0             17s
rook-ceph-exporter-minikube-5db8485649-8frcd        1/1     Running     0             17s
rook-ceph-mgr-a-69fbbd4fc7-c962k                    3/3     Running     0             53s
rook-ceph-mgr-b-798c486fd4-df68s                    3/3     Running     0             52s
rook-ceph-mon-a-7f5c676fd5-sf8tm                    2/2     Running     0             104s
rook-ceph-mon-b-7dbd7c889d-29c52                    2/2     Running     0             81s
rook-ceph-mon-c-67d7fd4549-lk5k8                    2/2     Running     0             70s
rook-ceph-operator-5d9594f9f8-slp6l                 1/1     Running     0             3m31s
rook-ceph-osd-0-cdd86ccd5-krldd                     1/2     Running     0             17s
rook-ceph-osd-prepare-minikube-k68f8                0/1     Completed   0             30s
```

Reply:
What do you see in the exporter pod logs? Do you also see this error?
```
2023-01-24T19:50:22.523+0000 7f6bb3d35f00 -1 asok(0x55d8ca57c560) AdminSocketConfigObs::init: failed: AdminSocket::bind_and_listen: failed to bind the UNIX domain socket to '/var/run/ceph/ceph-client.admin.asok': (17) File exists
```
Then a bunch of these lines:
```
system:0
```

Reply:
> 

Yeah, `system:0` lines are expected, it means server started and they repeat as every 10s it refreshes. I've tried finding a solution for getting rid of those lines, but well anyways it's related to exporter itself in this case.